x-pack/plugins/security_solution/common/endpoint/index_data.ts

```diff
@@ -98,3 +198,262 @@ async function indexAlerts(
+    name: policyName,
+    description: 'Protect the worlds data',
+    policy_id: agentPolicy.data.item.id,
+    enabled: true,
+    output_id: '',
+    inputs: [
+      {
+        type: 'endpoint',
+        enabled: true,
+        streams: [],
+        config: {
+          policy: {
+            value: policyConfigFactory(),
+          },
+        },
+      },
+    ],
+    namespace: 'default',
+    package: {
+      name: 'endpoint',
+      title: 'endpoint',
+      version: endpointPackageVersion,
+    },
+  };
+  const packagePolicy = (await kbnClient.request({
+    path: PACKAGE_POLICY_API_ROUTES.CREATE_PATTERN,
+    method: 'POST',
+    body: newPackagePolicyData,
+  })) as AxiosResponse<CreatePackagePolicyResponse>;
+  return packagePolicy.data.item;
+};
+
+const getEndpointPackageInfo = async (
+  kbnClient: KbnClient
+): Promise<GetPackagesResponse['response'][0]> => {
+  const endpointPackage = ((await kbnClient.request({
+    path: `${EPM_API_ROUTES.LIST_PATTERN}?category=security`,
+    method: 'GET',
+  })) as AxiosResponse<GetPackagesResponse>).data.response.find(
+    (epmPackage) => epmPackage.name === 'endpoint'
+  );
+
+  if (!endpointPackage) {
+    throw new Error('EPM Endpoint package was not found!');
+  }
+
+  return endpointPackage;
+};
+
+const fleetEnrollAgentForHost = async (
+  kbnClient: KbnClient & {
+    requestWithApiKey: (path: string, init?: RequestInit | undefined) => Promise<Response>;
+  },
+  endpointHost: HostMetadata,
+  agentPolicyId: string
+): Promise<undefined | PostAgentEnrollResponse['item']> => {
+  // Get Enrollement key for host's applied policy
+  const enrollmentApiKey = await kbnClient
+    .request<GetEnrollmentAPIKeysResponse>({
+      path: ENROLLMENT_API_KEY_ROUTES.LIST_PATTERN,
+      method: 'GET',
+      query: {
+        kuery: `fleet-enrollment-api-keys.policy_id:"${agentPolicyId}"`,
+      },
+    })
+    .then((apiKeysResponse) => {
+      const apiKey = apiKeysResponse.data.list[0];
+
+      if (!apiKey) {
+        return Promise.reject(
+          new Error(`no API enrollment key found for agent policy id ${agentPolicyId}`)
+        );
+      }
+
+      return kbnClient
+        .request<GetOneEnrollmentAPIKeyResponse>({
+          path: ENROLLMENT_API_KEY_ROUTES.INFO_PATTERN.replace('{keyId}', apiKey.id),
+          method: 'GET',
+        })
+        .catch((error) => {
+          // eslint-disable-next-line no-console
+          console.log('unable to retrieve enrollment api key for policy');
+          return Promise.reject(error);
+        });
+    })
+    .then((apiKeyDetailsResponse) => {
+      return apiKeyDetailsResponse.data.item.api_key;
+    })
+    .catch((error) => {
+      // eslint-disable-next-line no-console
+      console.error(error);
+      return '';
+    });
+
+  if (enrollmentApiKey.length === 0) {
+    return;
+  }
+
+  // Enroll an agent for the Host
+  const body: PostAgentEnrollRequest['body'] = {
+    type: 'PERMANENT',
+    metadata: {
```

Comment:
Here is a better `metadata` message, which will no longer cause the Fleet agent list to show `-` for host:

```javascript
{
    //...
    metadata: {
            local: {
                "elastic": {
                  "agent": {
                    "version": versionNumber
                  }
                },
                "host": {
                  "architecture": "x86_64",
                  "hostname": `artifact-downloader.${Date.now()}.elastic.co`,
                  "name": "artifact-downloader",
                  "id": "1c032ec0-3a94-4d54-9ad2-c5610c0eaba4",
                  "ip": [
                    "fe80::703b:b9e6:887d:7f5/64",
                    "10.0.2.15/24",
                    "::1/128",
                    "127.0.0.1/8"
                  ],
                  "mac": [
                    "08:00:27:d8:c5:c0"
                  ]
                },
                "os": {
                  "family": "windows",
                  "kernel": "10.0.19041.388 (WinBuild.160101.0800)",
                  "platform": "windows",
                  "version": "10.0",
                  "name": "Windows 10 Pro",
                  "full": "Windows 10 Pro(10.0)"
                }
            },
}
```

change `local.elastic.agent.version` to `8.0.0` (although, we should address that as well by using the kbnClient to query Kibana to get its version number, but we can do that another time (can you open an issue to track? :) )
Also - you will want to change the `host.hostname` ++ `host.name` to be `endpointHost.host`